pkg/skaffold/cluster/minikube.go

```diff
@@ -93,9 +103,46 @@ func minikubeExec(arg ...string) (*exec.Cmd, error) {
 	if err != nil {
 		return nil, fmt.Errorf("getting minikube executable: %w", err)
 	}
+
+	minVer, err := supportsUserFlagFunc()
+	if err != nil {
+		return nil, err
+	}
+	if minVer {
+		arg = append(arg, "--user=skaffold")
+	}
 	return exec.Command(b, arg...), nil
 }
 
+// versionCheck checks if the minikube version meet the minimum requirements for the user flag
+func supportsUserFlag() (bool, error) {
+	cmd := exec.Command("minikube", "version")
+	out, err := util.RunCmdOut(cmd)
+	if err != nil {
+		return false, err
+	}
+
+	currentVersion, err := getCurrentVersion(string(out))
+	if err != nil {
+		return false, err
+	}
```

Comment:
This code will cause `minikube version` to be executed each time we invoke minikube.  You need to use the cached currentVersion to determine whether to run `minikube version`.

Better yet, move this code into `minikubeBinary()`.